common/src/bytecode.rs

```diff
@@ -28,17 +28,18 @@ impl Bytecode {
             return Ok(Bytecode::default());
         }
 
-        // check that we start with 0x
-        if !s.starts_with("0x") {
-            return Err(BytecodeError::MissingHexPrefix);
-        }
         // and that the length is even
         if s.len() % 2 != 0 {
             return Err(BytecodeError::InvalidLength);
         }
 
+        // account for optional 0x prefix
+        let mut offset = 0;
+        if s.starts_with("0x") {
+            offset = 2;
+        }
```

Comment:
Could do something like
```rust
let s = s.strip_prefix("0x").unwrap_or(s);
```

Reply:
I should probably have been written with `strip_prefix` in the first place :see_no_evil: